MetagenomeAPI.spec

```diff
@@ -122,4 +122,74 @@ module MetagenomeAPI {
     funcdef get_annotated_metagenome_assembly(getAnnotatedMetagenomeAssemblyParams params)
         returns (getAnnotatedMetagenomeAssemblyOutput output) authentication required;
 
+    typedef structure {
+        string ref;
+        list<column_sorting> sort_by;
+        int start;
+        int limit;
+    } SearchOptions;
+
+    typedef structure {
+        string contig_id;
+        int start;
+        string strand;
+        int length;
+    } Location;
+
+    /*
+        aliases - mapping from alias name (key) to set of alias sources (value)
+        global_location - this is location-related properties that are
+            under sorting whereas items in "location" array are not,
+        feature_array - field recording which array a feature is located in
+            (features, mrnas, cdss, non_coding_features)
+        feature_idx - field keeping the position of feature in its array in a
+            Genome object,
+        ontology_terms - mapping from term ID (key) to term name (value).
+    */
+
+    typedef structure {
+        string feature_id;
+        mapping<string, list<string>> aliases;
+        string function;
+        list<Location> location;
+        string feature_type;
+        Location global_location;
+        string feature_array;
+        int feature_idx;
+        mapping<string, string> ontology_terms;
+    } FeatureData;
+
+    /*
+        num_found - number of all items found in query search (with only part of it returned in "features" list).
+    */
+
+    typedef structure {
+        string query;
+        int start;
+        list<FeatureData> features;
+        int num_found;
+    } SearchResult;
+
+    funcdef search(SearchOptions params) returns (SearchResult result) authentication optional;
+
+    typedef structure {
```

Comment:
Since these are the params to he function, it'd be nice to have comments on what each thing is.

Reply:
Or just tell people to refer to the readme